client/payment-details/summary/refund-modal/index.tsx

```diff
@@ -0,0 +1,127 @@
+/** @format **/
+
+/**
+ * External dependencies
+ */
+
+import React from 'react';
+import { Button, RadioControl } from '@wordpress/components';
+import { __, sprintf } from '@wordpress/i18n';
+import { useState } from '@wordpress/element';
+import interpolateComponents from '@automattic/interpolate-components';
+
+/**
+ * Internal dependencies.
+ */
+
+import ConfirmationModal from 'wcpay/components/confirmation-modal';
+import { Charge } from 'wcpay/types/charges';
+import { usePaymentIntentWithChargeFallback } from 'wcpay/data';
+import { PaymentChargeDetailsResponse } from 'wcpay/payment-details/types';
+import wcpayTracks from 'tracks';
+
+interface RefundModalProps {
+	charge: Charge;
+	formattedAmount: string;
+	onModalClose: () => void;
+}
+
+const RefundModal: React.FC< RefundModalProps > = ( {
+	charge,
+	formattedAmount,
+	onModalClose,
+} ) => {
+	const [ reason, setReason ] = useState< string | null >( null );
+
+	const [ isRefundInProgress, setIsRefundInProgress ] = useState< boolean >(
+		false
+	);
+
+	const { doRefund } = usePaymentIntentWithChargeFallback(
+		charge.payment_intent as string
+	) as PaymentChargeDetailsResponse;
+
+	const handleModalCancel = () => {
+		onModalClose();
+	};
+
+	const handleRefund = async () => {
+		wcpayTracks.recordEvent( 'payments_transactions_details_refund_full', {
+			payment_intent_id: charge.payment_intent,
+		} );
```

Comment:
Tracks data is limited to the merchants who have opted in for WooCommerce analytics. Given that we have tens of potential users per month, it could be that those who use the tool overlap very little with those who have analytics enabled. I suggest to keep tracks but to add a more reliable way of knowing that the refund was initiated through the transactions page, e.g. refund metadata.

Reply:
It looks like I missed some conversations, @dmvrtx - which analytics makes sense for this phase of projects? Are tracks on client tracks on the server or via meta-data? I'm asking if we can start with tracks and replace them later in a separate ticket with the desired output - currently, I'm not aware of specific expectations.

Reply:
We need metadata, because we might have only few data points and I'd rather have them all, than miss something due to the way Tracks work.

Reply:
@dmvrtx 
> refund metadata.

Are you referring to Stripe's metadata here?

Reply:
> Are you referring to Stripe's metadata here?

Yes. I'll check if we need to do anything on the server side to make it readily-available for querying, but I think it might be collected already.

Reply:
@dmvrtx It's a bit tricky. 

We now have three execution paths for how the refund can happen. 
1. The "old" from order page
2. (2.1) From the transaction page with order id
3. (2.2.) From the transactions page without order id

We can add the metadata to the request that we make to the server, but for the first two options, we call `wc_create_refund` which internally calls the request to the server, and it doesn't have any arguments that we can use to pass that informations in. We'd have to make changes in the WC core to make this possible ...

Reply:
@dmvrtx FYI, the meta would be added to the `refund` object in Stripe, not `payment intent` since the proxied endpoint is for `/refunds`

Reply:
Thanks. Since we send `refund_source` as part of the request to the server endpoint, it is possible to catch it there and it won't even require looking into the refund metadata.